Sources/Subs-Db-mysql.php

```diff
@@ -1099,4 +1099,20 @@ function smf_db_escape_string($string, $connection = null)
 	return mysqli_real_escape_string($connection === null ? $db_connection : $connection, $string);
 }
 
+/**
+ * Wrapper to handle null errors
+ *
+ * @param null|mysqli $connection = null The connection to use (null to use $db_connection)
+ * @return string escaped string
+ */
+function smf_db_errormsg($connection = null)
+{
+	global $db_connection;
+
+	if (is_null($connection) && is_null($db_connection))
```

Comment:
```suggestion
	if (is_null($connection) || is_null($db_connection))
```

Reply:
nvm

Reply:
So do I need to make the changes?

Reply:
Yes; thinking this over again, it is correct imo because it should check the provided connection before the global one. Also could we explicitly check? Another stupid suggestion, but matches everywhere else; also, I dislike superfluous function calls.

```suggestion
	if ($connection !== null || $db_connection !== null)
```

Reply:
I changed it, but it needs to be and, not or.  If either one is not null, it has a db connection and would just return a empty string if no error has occurred.

Reply:
oh oh did I really get those backwards? 😮 I accidentally inverted your logic, sorryyy.

Reply:
It happens to the best of us.

Reply:
looks like the inverted logic is in the commit